File: Week05/LeetCode-0150.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        S = []
        for token in tokens:
            if len(token)==1 and not token.isdigit():
                operand2 = S.pop()
                operand1 = S.pop()
                                
                if token == "+":
                    result = operand1 + operand2
                elif token == "-":
                    result = operand1 - operand2
                elif token == "*":
                    result = operand1 * operand2
                else: # token == "/"
                    result = int(operand1 / operand2)                
                S.append(result)                
            else:
                S.append(int(token))
        return S[0]

File: Week05/LeetCode-0150.cpp
class Solution {
public:
    ///----------------------------------------------------------
    /// Evaluate using a stack
    ///
    int evalRPN(vector<string>& tokens) {
        stack<int> S;

        for (auto token: tokens){
            if (token.length() == 1 && !isdigit(token[0])){
                // If this is an operator, then pop two operands
                // off the stack and apply the operator
                int operand2 = S.top(); S.pop();
                int operand1 = S.top(); S.pop();
                int result;

                switch(token[0]){
                    case '+': result = operand1 + operand2;
                        break;
                    case '-': result = operand1 - operand2;
                        break;
                    case '*': result = operand1 * operand2;
                        break;
                    case '/': result = operand1 / operand2;
                        break;
                } //end-switch

                S.push(result);

            } else {
                S.push(std::stoi(token));  // Convert the number in string to an integer
            } //end-else
        } //end-for

        return S.top();
    } // end-evalRPN
};